packages/editor-web-component/src/editor-web-component.tsx

```diff
@@ -0,0 +1,327 @@
+            },
+            {
+              type: 'math',
+              src: 'x_1',
+              inline: true,
+              children: [
+                {
+                  text: 'x_1',
+                },
+              ],
+            },
+            {
+              text: ' und ',
+            },
+            {
+              type: 'math',
+              src: 'x_2',
+              inline: true,
+              children: [
+                {
+                  text: 'x_2',
+                },
+              ],
+            },
+            {
+              text: ' frei wählen.',
+            },
+          ],
+        },
+      ],
+    },
+    {
+      plugin: 'geogebra',
+      state: 'nnrmthf4',
+    },
+  ],
+}
+
+// Could probably remove the export entirely, as the customElement is registered
+// below.
+export class EditorWebComponent extends HTMLElement {
+  private reactRoot: ReactDOM.Root | null = null
+  private container: HTMLDivElement
+
+  constructor() {
+    super()
+    // Create a shadow root for encapsulation. Right now, the css does not work
+    // with the shadow DOM.
+    // const shadowRoot = this.attachShadow({ mode: 'open' })
+    this.container = document.createElement('div')
+    // this.appendChild(container)
+    // shadowRoot.appendChild(container)
+
+    // fetch(cssUrl as string)
+    //   .then((response) => response.text())
+    //   .then((css) => {
+    //     const styleEl = document.createElement('style')
+    //     styleEl.textContent = css
+    //     shadowRoot.appendChild(styleEl)
+    //   })
+    //   .catch((error) =>
+    //     console.error('Error when trying to attach CSS string!', error)
+    //   )
+
+    // if (cssString && typeof cssString === 'string') {
+    //   const styleEl = document.createElement('style')
+    //   styleEl.textContent = cssString
+    //   shadowRoot.appendChild(styleEl)
+    // } else {
+    //   console.error('No CSS string could be found!', error)
+    // }
+
+    // this.reactRoot = ReactDOM.createRoot(container)
+  }
+
+  attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    console.log('AttributeChangedCallback', {
+      name,
+      oldValue,
+      newValue,
+    })
+    if (oldValue !== newValue) {
+      this.mountReactComponent()
+    }
+  }
+
+  static get observedAttributes() {
+    return ['initial-state']
+  }
+
+  connectedCallback() {
+    this.appendChild(this.container)
+
+    if (!this.reactRoot) {
+      this.reactRoot = ReactDOM.createRoot(this.container)
+    }
+
+    this.mountReactComponent()
+  }
+
+  mountReactComponent() {
+    const initialStateAttr = this.getAttribute('initial-state')
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const initialState: InitialState = initialStateAttr
+      ? (JSON.parse(initialStateAttr) as unknown as any)
+      : // TODO throw error here instead?
```

Comment:
Try / Catch ?

Will line 288 currently throw an error or always pass since it has the sample initial  state?

What is desired behaviour?